Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSReductionCorePreprocess.py

```diff
@@ -0,0 +1,136 @@
+
+from mantid.api import MatrixWorkspaceProperty, AlgorithmFactory, Progress
+from mantid.kernel import Direction
+from sans.algorithm_detail.mask_workspace import mask_bins
+from sans.common.enums import DetectorType
+
+from SANSReductionCoreBase import SANSReductionCoreBase
+
+
+class SANSReductionCorePreprocess(SANSReductionCoreBase):
+    def category(self):
+        return 'SANS\\Reduction'
+
+    def summary(self):
+        return 'Runs the initial core reduction elements. These are the steps which ' \
+               'can be carried out before event slicing.'
+
+    def PyInit(self):
+        # ----------
+        # INPUT
+        # ----------
+        # SANSReductionCorePreprocess has the same inputs as SANSReductionCore
+        self._pyinit_input()
+
+        # ----------
+        # OUTPUT
+        # ----------
+        self.declareProperty(MatrixWorkspaceProperty("OutputWorkspace", '', direction=Direction.Output),
+                             doc='The output workspace.')
+        self.declareProperty(MatrixWorkspaceProperty("DummyMaskWorkspace", '', direction=Direction.Output),
+                             doc='The histogram workspace which contains bin masks for non-compatibility mode.')
+
+        self.declareProperty(MatrixWorkspaceProperty("OutputMonitorWorkspace", '', direction=Direction.Output),
+                             doc='The output monitor workspace.')
+
+    def PyExec(self):
+        # Get the input
+        state = self._get_state()
+        state_serialized = state.property_manager
+        component_as_string = self.getProperty("Component").value
+        progress = self._get_progress()
+
+        # --------------------------------------------------------------------------------------------------------------
+        # 1. Crop workspace by detector name
+        #    This will create a reduced copy of the original workspace with only those spectra which are relevant
+        #    for this particular reduction.
+        # --------------------------------------------------------------------------------------------------------------
+        progress.report("Cropping ...")
+        workspace = self._get_cropped_workspace(component_as_string)
+
+        # --------------------------------------------------------------------------------------------
+        # 2. Perform dark run subtraction
+        #    This will subtract a dark background from the scatter workspace. Note that dark background subtraction
+        #    will also affect the transmission calculation later on.
+        # --------------------------------------------------------------------------------------------------------------
+
+        # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        # COMPATIBILITY BEGIN
+        # IMPORTANT: This section of the code should only be temporary. It allows us to convert to histogram
```

Comment:
has an issue been made to remove this temporary code once it is no longer necessary?

Reply:
#25899